pywinauto/controls/uia_controls.py

```diff
@@ -938,9 +938,27 @@ def __init__(self, elem):
         super(MenuItemWrapper, self).__init__(elem)
 
     # -----------------------------------------------------------
+    _cache_items = {}
     def items(self):
         """Find all items of the menu item"""
-        return self.children(control_type="MenuItem")
+        items = self.children(control_type="MenuItem")
+
+        # If a menu is not displayed then the UIA controls (children of self)
+        # will not yet exist. We can bring them into existence by clicking the
+        # menu (self). The controls can also (mysteriously) lose all their
+        # element_info. So once we see them bv clicking the menu we cache them
+        # for subsequent use.
+        if not items:
+            if self._cache_items.get(self.element_info.automation_id, None):
+                items = self._cache_items[self.element_info.automation_id].copy()
+            else:
+                self.click_input()
```

Comment:
`click_input()` is an invasive technique which can make `dump_tree()` a.k.a. `print_control_identifiers()` blinking with the menu. Item texts are taken into account by `best_match` algorithm so `dump_tree()` can call it to obtain all possible `best_match` names. As I remember, I already fixed similar problem for combo box wrapper pretty recently (I mean last year :) ).

Reply:
I'm not sure what you mean by "invasive" or what alternative there is.

Reply:
"Invasive" means that method `.dump_tree()` assumed as read-only. Think about it as a C++ `const` method. It calls `.items()` for every complicated element, but your implementation of `.items()` changes the state of the element (not a `const` method in terms of C++) and it even keeps submenu open. More right way to do it is expanding the submenu explicitly in your code, call what you need (list subitems) and then collapse the menu back if you don't need it any more.

Yes, some apps have better design of menu structure so the items are invisible but accessible. In your case a dynamic action is required to list the items.

Reply:
As per original note above: Part of me thinks the earlier solution I suggest of an optional argument click_to_find_children=False is impact neutral on existing functionality and does nothing special. But I'm loving the easy wrapper override facility that makes experimenting with solutions easy.